theatre/studio/src/panels/SequenceEditorPanel/RightOverlay/Marker.tsx

```diff
@@ -0,0 +1,83 @@
+import type {Pointer} from '@theatre/dataverse'
+import {useVal} from '@theatre/react'
+import {pointerEventsAutoInNormalMode} from '@theatre/studio/css'
+import getStudio from '@theatre/studio/getStudio'
+import {lockedCursorCssPropName} from '@theatre/studio/uiComponents/PointerEventsHandler'
+import useContextMenu from '@theatre/studio/uiComponents/simpleContextMenu/useContextMenu'
+import useRefAndState from '@theatre/studio/utils/useRefAndState'
+import React from 'react'
+import styled from 'styled-components'
+import {attributeNameThatLocksFramestamp} from '@theatre/studio/panels/SequenceEditorPanel/FrameStampPositionProvider'
+import type {SequenceEditorPanelLayout} from '@theatre/studio/panels/SequenceEditorPanel/layout/layout'
+
+const Mark = styled.div<{left: number}>`
+  position: absolute;
+  height: 20px;
+  width: 8px;
+  background-color: white;
+  z-index: 1;
+  color: black;
+  overflow: hidden;
+  font-size: 10px;
+  left: ${({left}) => left}px;
```

Comment:
Two issues:

1. The horizontal placement of this element will change every frame, every time the user pans/zooms. So we need to make sure it renders in an optimal way. Changing most css props like `left, top, width, margin, padding`  will cause the browser's rendering engine to re-layout and possibly re-paint this element and its parents/siblings and all their descendants. That's inefficient. Instead, you need to use `transform` so the browser can skip re-layout and re-paint, and likely offload the compositing to the GPU.
2. Changing props that you pass to styled-components will cause styled-components to create a new class name for each new permutation of all props values. This is fine for many props, but `left` will be changing once a frame per marker, so it can't be passed as a styled-component prop. Use `data-pos` or just set the `transform` directly from your react component like `<Mark style={{transform}}>`

Reply:
Cole had a comment about this above